internals/plan/plan.go

```diff
@@ -549,17 +549,22 @@ func (e *FormatError) Error() string {
 	return e.Message
 }
 
+// sectionAllocateOnce only allocates a section if it is nil, allowing on
+// demand allocation of sections only if it is present in the configuration.
+func sectionAllocateOnce[K comparable, V any](section map[K]V) map[K]V {
```

Comment:
The function name here is awkward, and more "how it may be used" rather than "what it does". Perhaps call it `makeMapIfNil`? Then the arg should be named something more generic too, maybe `m`, and the doc comment can probably just go away?

Reply:
Made naming changes as proposed. Resolving.